aten/src/ATen/native/SoftMax.cpp

```diff
@@ -446,6 +451,11 @@ Tensor log_softmax(const Tensor& input_, const int64_t dim_, c10::optional<Scala
   return result;
 }
 
+Tensor& log_softmax_out(const Tensor& input_, const int64_t dim_, c10::optional<ScalarType> dtype, Tensor& output_) {
+  Tensor converted = dtype.has_value() ? input_.toType(dtype.value()) : input_;
+  return at::_log_softmax_out(output_, converted, dim_, false);
```

Comment:
is there any reason we can't just promote these private API's to be public, instead of leaving them?

Reply:
I believe this is more of a short term fix to unblock our new edge runtime, but I think consolidating log_softmax with _log_softmax is probably a better solution. cc @gmagogsfm  to confirm